opm/output/eclipse/WriteInit.cpp

```diff
@@ -375,6 +473,47 @@ namespace {
         initFile.write("DZ"   , dz);
     }
 
+    void writeGridGeometryLGRCell(const ::Opm::EclipseGrid&         grid,
+                                  const ::Opm::EclipseGridLGR&      lgr_grid,
+                                  const ::Opm::UnitSystem&          units,
+                                        ::Opm::EclIO::OutputStream::Init& initFile,
+                                  const                  int        nx,
+                                  const                  int        ny,
+                                  const                  int        nz)
+    {
+        const auto length = ::Opm::UnitSystem::measure::length;
+        auto convert_length = [&units](std::vector<float>& depth) 
+        {
+            for (std::size_t index = 0; index < depth.size(); ++index) {
+                depth[index] = units.from_si(length, depth[index]);
+            }           
```

Comment:
This is just
```C++
std::transform(depth.begin(), depth.end(), depth.begin(), [&units](const auto d) { return units.from_si(length, d); });
```
and seeing as the function is used exactly once in this scope I suggest just using that `transform()` call in-place.